src/plugins/twitter/client.py

```diff
@@ -0,0 +1,141 @@
+from datetime import datetime, timedelta
+import requests
+from requests_oauthlib import OAuth1
+
+
+class TwitterClient:
+
+    """ Twitter client for searching tweets about DOIs.
+
+    Please note, this uses the Premium Twitter search API. For more details
+    visit https://developer.twitter.com/en/docs/tweets/search/api-reference
+    /premium-search.html
+    """
+
+    verify_url = 'https://api.twitter.com/1.1/account/verify_credentials.json'
```

Comment:
I feel this should be retrieved from some kind of config.

Reply:
Yeah, it's one of those things I'm never really sure about... 

The way I see it, this URL isn't really something that should be 'setable'. Generally speaking, it shouldn't really change, unless Twitter updates its API endpoints, in which case we will need to update the client. If someone else decides to use this, they should be able to just update the client and it should work. It shouldn't fail just because they haven't updated their own config files - but again, I'm not sure if this is the correct way to view this.

Reply:
Yeah, I can see the sense in doing it like you are. This way, you're concealing the details of which Twitter API endpoints are consumed from the user.